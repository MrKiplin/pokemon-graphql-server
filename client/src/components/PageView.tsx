import React, { Fragment } from "react";
import { Helmet } from "react-helmet";
import { Container } from "semantic-ui-react";
import PageHeader from "./PageHeader";

interface PageViewProps {
  title: string;
}

const PageView: React.FC<PageViewProps> = ({ title, children }) => (
  <Fragment>
    <Helmet titleTemplate="Pokemon GraphQL Server">
      <title>{title}</title>
    </Helmet>
    <Container>
      <PageHeader
        header="Pokemon GraphQL Server"
        subHeader="Simple GraphQL stack for connecting to the Pokemon REST API"
      />
      {children}
    </Container>
  </Fragment>
);

export default PageView;
