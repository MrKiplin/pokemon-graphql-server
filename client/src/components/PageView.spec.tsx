import { mount } from "enzyme";
import PageView from "./PageView";

describe("PageView", () => {
  it("Should match props and snapshot", () => {
    const component = mount(<PageView {...{ title: "test-title" }} />);

    expect(component).toMatchSnapshot();
    expect(component.find("PageView").prop("title")).toEqual("test-title");
  });
});
