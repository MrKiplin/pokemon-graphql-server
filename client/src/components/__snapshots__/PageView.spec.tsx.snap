// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageView Should match props and snapshot 1`] = `ReactWrapper {}`;
